lib/workers/bin/provider_worker

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env ruby
+
+require File.expand_path("../../../config/environment", __dir__)
+
+ems_id = ENV["EMS_ID"]
+
+ems      = ExtManagementSystem.find(ems_id)
+all_emss = [ems] + ems.child_managers
+ems_opts = all_emss.map do |e|
+  e.attributes.merge(
+    "endpoints"       => e.endpoints,
+    "authentications" => e.authentications
+  )
+end
+
+provider_opts = {
+  :ems       => ems_opts,
+  :messaging => MiqQueue.messaging_client_options,
+  :settings  => Settings.to_h
+}
+
+# Create a temp file and immediately unlink it to create a
+# secure hidden file to be used for the child process' stdin.
+stdin_tmp = Tempfile.new
+File.unlink(stdin_tmp.path)
+
+stdin_tmp.write(provider_opts.to_json)
+stdin_tmp.rewind
+
+$stdin.reopen(stdin_tmp)
```

Comment:
Once the other side reads it, what happens to the file?  Does it just stick around forever?  Is it still readable by other processes (I know it's unlinked, but wondering if you can get the inode or something)?

Reply:
> Once the other side reads it, what happens to the file? Does it just stick around forever?

Since the file is unlinked the only reference to it is the one we are holding here, so when the process exits that refcount drops to 0 and the inode is deleted.

> Is it still readable by other processes (I know it's unlinked, but wondering if you can get the inode or something)?

It is not possible to open a file by inode only, since doing so would violate all filesystem access control